internal/gh/rest.go

```diff
@@ -0,0 +1,218 @@
+package gh
+
+import (
+	"encoding/json"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"regexp"
+)
+
+const baseURL = "https://api.github.com"
+
+type Result struct {
+	Entry string
+}
+
+// Create an HTTP request using the parameters, token, method and http client.
+func createHttpRequest(client *http.Client, reqUrl, requestMethod string, requestBody io.ReadCloser, queryParameters, headerParameters map[string]string) (*http.Request, error) {
+
+	req, err := http.NewRequest(requestMethod, reqUrl, requestBody)
+	if err != nil {
+		return req, err
+	}
+
+	for key, value := range headerParameters {
+		req.Header.Set(key, value)
+	}
+
+	query := req.URL.Query()
+
+	for key, value := range queryParameters {
+		query.Add(key, value)
+	}
+
+	req.URL.RawQuery = query.Encode()
+
+	return req, nil
+}
+
+// Update an existing HTTP request using the parameters, token and method.
+func updateHttpRequest(reqUrl, requestMethod string, requestBody io.ReadCloser, queryParameters, headerParameters map[string]string, req *http.Request) error {
+
+	parsedUrl, err := url.Parse(reqUrl)
+	if err != nil {
+		slog.Default().Error("Error while parsing the URL", err)
```

Comment:
the syntax for the error log is a bit different. you give a message to log, then you provide key value pairs for further logging.
the key is the name of whatever you want to log, in this case something like "error" and the value can be any struct or whatever, which is then stringified and printed after the key.


```suggestion
		slog.Default().Error("Error while parsing the URL", "error", err)
```